lib/src/mixins/batched_redraws.dart

```diff
@@ -6,23 +6,42 @@ import 'dart:html';
 import 'package:react/react.dart' as react;
 
 class _RedrawScheduler implements Function {
-  Set<react.Component> _components = new Set();
+  Map<react.Component, List<Function>> _components =
+      <react.Component, List<Function>>{};
 
-  void call(react.Component component) {
+  void call(react.Component component, [callback()]) {
     if (_components.isEmpty) {
       _tick();
     }
-    _components.add(component);
+
+    _components[component] ??= [];
+
+    _components[component].add(callback);
   }
 
   Future _tick() async {
     await window.animationFrame;
     _components
-      ..forEach((c) {
-        c.setState({});
+      ..forEach((component, callbacks) {
+        var chainedCallbacks = callbacks.fold(null, _chain) ?? _noop;
```

Comment:
Could this be simplified to something like below?

```dart
var chainedCallbacks = () {
  if (callbacks.isNotEmpty) {
    callbacks.forEach((callback) => callback());
  }
}
```

And then on line 19 above we could add the callback only if it isn't `null`. This would eliminate the `_noop` and `_chain` functions.